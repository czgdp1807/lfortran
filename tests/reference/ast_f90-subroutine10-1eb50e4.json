{
    "basename": "ast_f90-subroutine10-1eb50e4",
    "cmd": "lfortran --show-ast-f90 --no-color {infile}",
    "infile": "tests/subroutine10.f90",
    "infile_hash": "de16e99a96162482992b5d88a4d803f4389d34aa6fc5809a068f8d36",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_f90-subroutine10-1eb50e4.stdout",
    "stdout_hash": "33210b065e12eec5269a090b4451aea86dac683c032a35d2fe36141a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}