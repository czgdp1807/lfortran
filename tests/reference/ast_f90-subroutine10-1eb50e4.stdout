program test
real :: x

call sub(x, 1, *100, *200)
print *, "Success:", x
stop

100 print *, "Negative input value"
stop

200 print *, "Input value too large"
stop

end program test


subroutine sub(x, i, *, *)
real, intent(out) :: x
integer, intent(in) :: i
if (i < 0) then
    return 1
end if
if (i > 10) then
    return 2
end if
x = i
end subroutine sub
